tests/item_test.cpp

```diff
@@ -1098,3 +1101,101 @@ TEST_CASE( "item_rotten_contents", "[item]" )
     //   butter
     CHECK( wrapper.get_category_of_contents().id == item_category_food );
 }
+
+
+TEST_CASE( "uncraft_sanity_check", "[item]" )
+{
+    std::vector<const itype *> all_items = item_controller->all();
+
+    // only allow so many failures before stopping
+    int number_of_failures = 0;
+
+    // 10% in either side should be fine
+    const float approximation = 0.1f;
+
+    for( const itype *type : all_items ) {
+        const item target( type, calendar::turn_zero, item::solitary_tag{} );
+
+        // if comestible, gun or book, skip, but if still has uncraft, do not skip
+        if( ( type->comestible || type->gun || type->book ) && target.get_uncraft_components().empty() ) {
+            continue;
+        }
+
+        if( type->src.back().second == MOD_INFORMATION_test_data ) {
+            continue;
+        }
+
+        const units::mass item_weight = type->weight;
+
+        // ethereal, technical, mutation related item etc
+        if( item_weight == 0_gram ||
+            type->volume == 0_ml ||
+            type->has_flag( flag_INTEGRATED ) ||
+            type->has_flag( flag_NO_TAKEOFF ) ||
+            type->has_flag( flag_AURA ) ||
+            type->has_flag( flag_PERSONAL ) ) {
+
+            continue;
+        }
+
+        const bool not_in_list = test_data::known_bad_uncraft.count( type->id ) == 0;
```

Comment:
The loop should just continue if the item is already on the list.

Reply:
it does continue tho, this just stashes the check in a bool that we use later to print the message or not?